daemon/cluster/networks.go

```diff
@@ -87,9 +87,14 @@ func (c *Cluster) UpdateAttachment(target, containerID string, config *network.N
 	if !ok || attacher == nil {
 		return fmt.Errorf("could not find attacher for container %s to network %s", containerID, target)
 	}
+	if attacher.config != nil {
```

Comment:
This looks racy. We might check this before the other goroutine sets `attacher.config`. Then we'll block forever trying to send to the channel.